search/search_repository_blackbox_test.go

```diff
@@ -585,6 +585,46 @@ func (s *searchRepositoryBlackboxTest) TestFilter() {
 		})
 	})
 
+	s.T().Run("fail - with incorrect value type", func(t *testing.T) {
+		// given
+		t.Run("integer instead of UUID", func(t *testing.T) {
+			filter := `{"space": 123}`
+			_, count, _, _, err := s.searchRepo.Filter(context.Background(), filter, nil, nil, nil)
+			require.Error(t, err)
+			assert.Equal(t, 0, count)
+		})
+
+		t.Run("string instead of UUID", func(t *testing.T) {
+			filter := `{"space": "foo"}`
+			_, count, _, _, err := s.searchRepo.Filter(context.Background(), filter, nil, nil, nil)
+			require.Error(t, err)
+			assert.Equal(t, 0, count)
+
+		})
+
+		// Regression test for https://github.com/openshiftio/openshift.io/issues/4429
+		t.Run("string instead of integer", func(t *testing.T) {
+			filter := `{"number":{"$EQ":"asd"}`
```

Comment:
The curly bracket is not closed. The assertion should check for the message.

Reply:
Thank you. I've fixed it https://github.com/fabric8-services/fabric8-wit/pull/2321/commits/5f559bc7482bda27f9dae64b838f40d0edac27df

Reply:
The error message depends on the input value. Do you think we should still do it?
Error for an integer would be different from the error for a string.